csharp/tools/tracing-config.lua

```diff
@@ -1,4 +1,50 @@
 function RegisterExtractorPack(id)
+    function requiresQuoting(s)
+        return s ~= nil and s ~= '' and string.find(s, "[ \t\n\v\"]")
+    end
+
+    function insertMany(t, c, n)
+        for i = 1, n do
+            table.insert(t, c)
+        end
+    end
+
+    -- This implementation is copied from
+    -- https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
```

Comment:
This looks good to me, but I think we should move this function to the base.lua library near the definition of NativeArgumentsToArgv. The functions NativeArgumentsToArgv and QuoteArg are roughly each others inverse: `NativeArgumentsToArgv(args.map(QuoteArg).join(' ')) == args`